float64 magnitude
float64 phase
